protobuf/protoeditor-jvm/src/com/intellij/protobuf/jvm/names/NameGeneratorContributor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.protobuf.jvm.names;
+
+import com.google.common.collect.ImmutableList;
+import com.intellij.protobuf.lang.psi.*;
+import com.intellij.openapi.extensions.ExtensionPointName;
+import java.util.Optional;
+
+/** Extension point for things that provide names for a protobuf generated class. */
+public interface NameGeneratorContributor {
+  ExtensionPointName<NameGeneratorContributor> EP_NAME =
+    ExtensionPointName.create("com.intellij.protobuf.jvm.nameGeneratorContributor");
+
+  /**
+   * Allows a generator provider to opt into being used for a given protobuf file.
+   *
+   * @param file the protobuf file to provide name generators for
+   * @return boolean indicating whether or not the extension will process the given file
+   */
+  boolean isApplicable(PbFile file);
+
+  /**
+   * Provides name generators for fields contained in the given protobuf file.
+   *
+   * <p><b>Warning:</b> This method should be called in a read action compatible context.
+   *
+   * @param file the protobuf file to provide name generators for
+   * @return a list of name generators for the provided file
+   */
+  ImmutableList<NameGenerator> contributeGenerators(PbFile file);
+
+  static Optional<Integer> parseIntOption(PbOptionExpression optionExpression) {
+    return Optional.ofNullable(optionExpression.getNumberValue())
+      .map(PbNumberValue::getLongValue)
+      .map(Long::intValue);
+  }
+
+  static Optional<String> parseStringOption(PbOptionExpression optionExpression) {
+    return Optional.ofNullable(optionExpression.getStringValue())
+      .map(ProtoStringValue::getAsString);
+  }
+
+  static Optional<Boolean> parseBoolOption(PbOptionExpression optionExpression) {
+    return Optional.ofNullable(optionExpression.getIdentifierValue())
+      .map(PbIdentifierValue::getBooleanValue);
+  }
+}
```

Comment:
Should we extract these utility methods to some new/existing `*Utils` class? For me it looks better when EP interface does contain only methods intended to be implemented

Reply:
Extracted these out to com.intellij.protobuf.lang.names.NameGeneratorUtils